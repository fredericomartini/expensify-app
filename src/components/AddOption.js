import React, { Component } from 'react';

export default class AddOption extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: undefined,
    };

    this.handleAddOne = this.handleAddOne.bind(this);
  }

  handleAddOne(event) {
    event.preventDefault();
    const {
      target: {
        elements: { option },
      },
    } = event;

    const error = this.props.handleAddOne(option.value);
    this.setState(() => ({ error }));
    option.value = '';
  }

  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.handleAddOne}>
          <input type="text" name="option" />
          <button>Add Option</button>
        </form>
      </div>
    );
  }
}
