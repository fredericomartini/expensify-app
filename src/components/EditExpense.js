import React from 'react';

const EditExpense = (props) => {
  const {
    match: {
      params: { id },
    },
  } = props;
  console.log(props);
  return (
    <div>
      This is the edit expense for the item:
      {id}
    </div>
  );
};

export default EditExpense;
