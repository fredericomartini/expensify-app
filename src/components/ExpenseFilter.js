import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter } from '../actions/filters';

const ExpenseFilter = (props) => (
  <div>
    Search for:
    {' '}
    <input
      type="text"
      value={props.filters.text}
      onChange={(e) => {
        props.dispatch(setTextFilter(e.target.value));
      }}
    />
  </div>
);

// Maps estado do store para componente props
const mapStateToProps = ({ filters }) => ({
  filters,
});

export default connect(mapStateToProps)(ExpenseFilter);
