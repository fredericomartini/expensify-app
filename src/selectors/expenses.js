import moment from 'moment';

// Visible Expenses
export default (expenses, {
  text, sortBy, startDate, endDate,
}) => expenses
  .filter((expense) => {
    const createdAtMoment = moment(expense.createdAt);
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
    const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
    return startDateMatch && endDateMatch && textMatch;
  })
  .sort((a, b) => {
    if (sortBy === 'date') {
      return moment(a.createdAt).isBefore(moment(b.createdAt)) || -1;
    }
    if (sortBy === 'amount') {
      return a.amount < b.amount || -1;
    }
    return [b, a];
  });
