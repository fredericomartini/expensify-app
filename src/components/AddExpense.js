import React from 'react';
import { connect } from 'react-redux';
import { addExpense } from '../actions/expenses';

const AddExpense = (props) => (
  <div>
    <h3>This is the add expense</h3>
    Description:
    {' '}
    <input name="description" type="text" />
    Amount:
    {' '}
    <input name="amount" type="text" />
    <button
      type="submit"
      onClick={(e) => {
        console.log(props);
        // props.dispatch(addExpense())
      }}
    >
      Save Expense
    </button>
  </div>
);

const mapStateToProps = (state) => ({ ...state });

export default connect(mapStateToProps)(AddExpense);
