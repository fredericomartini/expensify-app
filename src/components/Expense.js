import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

const Expense = ({
  dispatch, id, description, amount, createdAt,
}) => (
  <div>
    <h3>{description}</h3>
    <p>
      Amount:
      {' '}
      {amount}
      {' '}
createdAt:
      {createdAt}
    </p>
    <button
      onClick={() => {
        dispatch(removeExpense({ id }));
      }}
    >
      Remove
    </button>
  </div>
);

export default connect()(Expense);
