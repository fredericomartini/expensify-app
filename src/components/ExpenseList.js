import React from 'react';
import { connect } from 'react-redux';
import Expense from './Expense';
import expensesSelector from '../selectors/expenses';
import ExpenseFilter from './ExpenseFilter';
import { sortByAmount, sortByDate } from '../actions/filters';

const ExpenseList = (props) => (
  <div>
    <h1>Expenses List</h1>
    <ExpenseFilter />
    <h3>
      SortBy:
      <select
        value={props.sortBy || 'date'}
        defaultValue="date"
        onChange={({ target: { value } }) => {
          switch (value) {
            case 'amount':
              return props.dispatch(sortByAmount());
            default:
              return props.dispatch(sortByDate());
          }
        }}
      >
        <option value="date">Date</option>
        <option value="amount">Amount</option>
      </select>
    </h3>
    {props.expenses && props.expenses.length !== 0 ? (
      props.expenses.map((expense) => <Expense key={expense.id} {...expense} />)
    ) : (
      <p> No expenses</p>
    )}
  </div>
);

// Maps estado do store para componente props
const mapStateToProps = (state) => ({
  expenses: expensesSelector(state.expenses, state.filters),
  sortBy: state.filters.sortBy,
});

export default connect(mapStateToProps)(ExpenseList);
