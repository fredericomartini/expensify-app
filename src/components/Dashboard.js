import React from 'react';
import ExpenseFilter from './ExpenseFilter';
import ExpenseList from './ExpenseList';

const Dashboard = () => (
  <div>
    <ExpenseFilter />
    <ExpenseList />
  </div>
);

export default Dashboard;
