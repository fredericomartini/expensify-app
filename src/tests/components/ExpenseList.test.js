import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('should RENDER EXPENSE LIST WITH EXPENSES', () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test('should RENDER EXPENSE LIST WITH NO EXPENSES', () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});
