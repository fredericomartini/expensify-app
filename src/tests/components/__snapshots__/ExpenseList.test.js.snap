// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should RENDER EXPENSE LIST WITH EXPENSES 1`] = `
<div>
  <h1>
    Expenses List
  </h1>
  <h3>
    SortBy:
    <select
      onChange={[Function]}
      value="date"
    >
      <option
        value="date"
      >
        Date
      </option>
      <option
        value="amount"
      >
        Amount
      </option>
    </select>
  </h3>
  <Component
    expense={
      Object {
        "amount": 199,
        "createdAt": "1970-01-01T00:00:00.000Z",
        "description": "Watter bill",
        "id": "1",
        "note": "",
      }
    }
    key="1"
  />
  <Component
    expense={
      Object {
        "amount": 4500,
        "createdAt": "1969-12-28T00:00:00.000Z",
        "description": "Bank accounts",
        "id": "2",
        "note": "",
      }
    }
    key="2"
  />
  <Component
    expense={
      Object {
        "amount": 325,
        "createdAt": "1970-01-05T00:00:00.000Z",
        "description": "Gas bill",
        "id": "3",
        "note": "",
      }
    }
    key="3"
  />
</div>
`;

exports[`should RENDER EXPENSE LIST WITH NO EXPENSES 1`] = `
<div>
  <h1>
    Expenses List
  </h1>
  <h3>
    SortBy:
    <select
      onChange={[Function]}
      value="date"
    >
      <option
        value="date"
      >
        Date
      </option>
      <option
        value="amount"
      >
        Amount
      </option>
    </select>
  </h3>
  <p>
     No expenses
  </p>
</div>
`;
